### PR TITLE
add module odgi/sort

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

This PR progresses https://github.com/nf-core/pangenome/issues/70.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/odgi/sort/main.nf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/odgi:0.8.0--py310hc8f18ef_0':
+        'quay.io/biocontainers/odgi:0.8.0--py310hc8f18ef_0' }"
+
+    input:
+    tuple val(meta), path(graph)
+
+    output:
+    tuple val(meta), path("*.og")                 , emit: sorted_graph
+    path "versions.yml"                           , emit: versions
```
Comment (JoseEspinosa) [inline modules/nf-core/odgi/sort/main.nf:15]:
```suggestion
    tuple val(meta), path("*.og"), emit: sorted_graph
    path "versions.yml"          , emit: versions
```

File: modules/nf-core/odgi/sort/main.nf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    odgi \\
+        sort \\
+        --threads $task.cpus \\
+        --idx ${graph} \\
+        --out ${prefix}.og \\
+        $args
```
Comment (JoseEspinosa) [inline modules/nf-core/odgi/sort/main.nf:29]:
```suggestion
        $args
        
```

File: modules/nf-core/odgi/sort/meta.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - graph:
+      type: file
+      description: Pangenome graph in GFA v1.0 format or in ODGI binary format
+      pattern: "*.{gfa,og}"
+
+## TODO nf-core: Add a description of all of the variables used as output
```
Comment (JoseEspinosa) [inline modules/nf-core/odgi/sort/meta.yml:28]:
```suggestion
```

Comment (nvnieuwk) [inline modules/nf-core/odgi/sort/meta.yml:28]:
```suggestion
```

File: tests/modules/nf-core/odgi/sort/test.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+- name: "odgi sort"
+  command: nextflow run ./tests/modules/nf-core/odgi/sort -entry test_odgi_sort -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/odgi/sort/nextflow.config
+  tags:
+    - "odgi"
+    - "odgi/sort"
+  files:
+    - path: "output/odgi/test.og"
+    - path: "output/odgi/versions.yml"
```
Comment (JoseEspinosa) [inline tests/modules/nf-core/odgi/sort/test.yml:8]: No need to test for the `versions.yml` file
```suggestion
```

File: tests/modules/nf-core/odgi/sort/test.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+      md5sum: e5e833e170b1f85c26147e9b6228bceb
```
Comment (nvnieuwk) [inline tests/modules/nf-core/odgi/sort/test.yml:9]:
```suggestion
```

**Review by nvnieuwk — APPROVED** [review]
LGTM (well this one was a lot easier than the last one :grin: )

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
LGTM! 
Just requested more minor changes mostly stylilstic

File: tests/modules/nf-core/odgi/sort/test.yml (hunk 4 above)
Comment (nvnieuwk) [reply tests/modules/nf-core/odgi/sort/test.yml:8]: Apparently only the md5sum check for `versions.yml` needs to be removed (see https://nfcore.slack.com/archives/CE5LG7WMB/p1666265866115999)

**Comment by subwaystation** [comment]
@nvnieuwk @JoseEspinosa I think I addressed all of your comments. One final approval and thanks to you!

**Review by nvnieuwk — APPROVED** [review]

File: tests/modules/nf-core/odgi/sort/test.yml (hunk 4 above)
Comment (JoseEspinosa) [reply tests/modules/nf-core/odgi/sort/test.yml:8]: OK, thanks @nvnieuwk good to know I started to delete it in my reviews from this [comment](https://github.com/nf-core/modules/pull/2328#discussion_r999065900) 😆

Comment (nvnieuwk) [reply tests/modules/nf-core/odgi/sort/test.yml:8]: No problem, I've been doing that for a while too now. That was apparently not right :grin:

**Pull request merged**